### PR TITLE
update tests and asciidocs

### PR DESCRIPTION
**Pull request by jo2**

### CODE REVIEW

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 1]
```diff
@@ -85,10 +88,8 @@ To access the functionality of coderadar you have to register a user.
 You need to define a username and a password. The password will be sent as
 plain text and hashed on server side for persisting.
 
-==== Hypermedia links
-include::{snippets}/user/registration/links.adoc[]
-
 ==== Registering a User
+// TODO should password really be send as plaintext or is it in plaintext but encrypted?
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: The password is sent as plaintext over HTTPS, so yes, in a way, everything we send to the server should be encrypted.

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 2]
```diff
@@ -121,6 +122,8 @@ If the log in is successful, user obtains two JSON Web Tokens https://jwt.io
 
 How to use the tokens is described in <<authentication, Authentication>>
 
+// TODO shouldn't it be a hash which is sent to the server?
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: If you're talking about the password, then only the server does the hashing, doing so in the front-end would require that we use the same algorithm as  in the backend.  Or do you propose hashing it twice? Once in the front-end and once in the back-end?

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 3]
```diff
@@ -325,40 +333,28 @@ include::{snippets}/modules/delete/http-response.adoc[]
...
-is supposed to contain all source code files of your project.
-
-File patterns may either define a pattern for files to be INCLUDED or files to be EXCLUDED. To compile the list of files
-contained in a certain file set (like the SOURCE file set), coderadar will first apply all INCLUDE patterns of that file set and then apply all EXCLUDE
-patterns. Thus, the file set will include all INCLUDED files except those that were part of an EXCLUDE pattern.
+A File Pattern describes a set of files within a Project's code base. Each project can have several file patterns
+defined. File Patterns may either define a pattern for files to be INCLUDED or files to be EXCLUDED.
 
-If a project has no file patterns defined, no files will be analyzed.
+If a project has no File Patterns defined, no files will be analyzed.
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: I think saying the analysis won't start at all, would be more accurate here.

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 4]
```diff
@@ -432,141 +454,49 @@ include::{snippets}/analyzerConfiguration/getSingle/http-request.adoc[]
...
-By providing an Analyzing Job to coderadar, you can trigger an analysis of a Project's codebase. Note
-the following hints:
-
-* coderadar will automatically start to analyze the codebase once an Analyzing Job has been provided,
-unless the Analyzing Job is set to inactive.
-* Only one Analyzing Job can be active for a project at a time. You will have to wait until a running job
-is finished to start another one.
-* New incoming commits to the underlying git repository will automatically be analyzed, unless the Analyzing
-Job is set to inactive.
+Analyzing Jobs can be started for a project in coderadar. By default a Project's codebase is not analyzed.
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: I'd replace Analyzing Jobs simply with analysis, as we no longer have "jobs".

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 5]
```diff
@@ -432,141 +454,49 @@ include::{snippets}/analyzerConfiguration/getSingle/http-request.adoc[]
...
-==== Structure
-
-include::{snippets}/analyzing-job/create-update/request-fields.adoc[]
-
-==== Hypermedia Links
-
-include::{snippets}/analyzing-job/create-update/links.adoc[]
+[NOTE]
+The execution of an Analyzing Job may take some time. This is especially true for larger projects.
+Spring Boot for example takes about 30 minutes. This may vary with different hardware specifications.
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: I think we can put such information into the admin guide.

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 6]
```diff
@@ -603,9 +533,9 @@ You can list all metrics that have been measured for your project by calling the
 
 [NOTE]
 =====
-Metrics will become available as soon as you have provided an <<creating-analyzing-job,Analyzing Job>> for your project.
-At that moment, analysis of each commit defined in the job starts. Depending on how many commits that are,
-analysis will take a while.
+Metric values will become available as soon as an <<start-analyzing-job,Analyzing Job>> starts for your project.
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: Once again, we can replace Analyzing Job with analysis.

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 7]
```diff
@@ -603,9 +533,9 @@ You can list all metrics that have been measured for your project by calling the
...
+At that moment, analysis of each commit defined in the job starts. Depending on how many commits that are and how
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: Commits aren't actually defined in the "job" though. Maybe a few words about how only commits in the date-range set for the project will be analyzed.

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 8]
```diff
@@ -739,15 +631,18 @@ Also, the tree contains information about files that have been renamed or moved
...
 include::{snippets}/metrics/deltaTree/request-fields.adoc[]
 
 ===== Response Structure
+// TODO renamedFrom, renamedTo, changes without data type; renamedFrom, renamedTo without description
 
 include::{snippets}/metrics/deltaTree/response-fields.adoc[]
 
 ===== Example Request
 
+// TODO still valid?
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: yes

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 9]
```diff
@@ -739,15 +631,18 @@ Also, the tree contains information about files that have been renamed or moved
 commits.
 
 ===== Request Structure
+// TODO encoding in commit2
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: What do you mean by encoding?

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 10]
```diff
@@ -722,6 +613,7 @@ include::{snippets}/metrics/tree/response-fields.adoc[]
 
 ===== Example Request
 
+// TODO still valid?
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: Currently only get works, but both should be available. So you can leave the note for now.

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 11]
```diff
@@ -659,6 +586,7 @@ include::{snippets}/metrics/commit/metrics/request-fields.adoc[]
...
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: yes

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 12]
```diff
@@ -39,6 +41,7 @@ The following table describes how the coderadar API interprets the HTTP verbs.
 
 |PUT
 |PUT is not used at this time.
+// TODO PUT really not used?
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: no, not used

File: coderadar-test/src/main/asciidoc/restapi.adoc [hunk 13]
```diff
@@ -53,6 +56,7 @@ provided as parameters encoded in the URL. However, in some cases parameters are
 encode them in the URL. In these cases the parameters are expected as JSON string within the request
 body. Since some tools don't allow GET requests with a body, coderadar accepts the POST method in
 these cases as well.
+// TODO still true?
```
Comment (maximAtanasov) [inline coderadar-test/src/main/asciidoc/restapi.adoc]: yes

File: coderadar-test/src/test/java/io/reflectoring/coderadar/rest/query/GetMetricValuesOfTwoCommitsControllerTest.java [hunk 14]
```diff
@@ -0,0 +1,180 @@
+package io.reflectoring.coderadar.rest.query;
```
Comment (maximAtanasov) [inline coderadar-test/src/test/java/io/reflectoring/coderadar/rest/query/GetMetricValuesOfTwoCommitsControllerTest.java]: This file was the same as GetDeltaTreeForTwoCommitsControllerTest so I've deleted it when merging with master.

**Comment by maximAtanasov** [comment]
I also get warnings for missing files when building the docs:

```
asciidoctor: WARNING: restapi.adoc: line 75: include file not found: C:/Users/atanasov/workspace/CODE_RADAR/update-asciidoc/coderadar/coderadar-test/build/generated-snippets/projects/create/error400/http-response.adoc
asciidoctor: WARNING: restapi.adoc: line 360: include file not found: C:/Users/atanasov/workspace/CODE_RADAR/update-asciidoc/coderadar/coderadar-test/build/generated-snippets/filepatterns/create-update/http-request.adoc
asciidoctor: WARNING: restapi.adoc: line 364: include file not found: C:/Users/atanasov/workspace/CODE_RADAR/update-asciidoc/coderadar/coderadar-test/build/generated-snippets/filepatterns/create-update/http-response.adoc
asciidoctor: WARNING: restapi.adoc: line 608: include file not found: C:/Users/atanasov/workspace/CODE_RADAR/update-asciidoc/coderadar/coderadar-test/build/generated-snippets/metrics/tree/request-fields.adoc
asciidoctor: WARNING: restapi.adoc: line 636: include file not found: C:/Users/atanasov/workspace/CODE_RADAR/update-asciidoc/coderadar/coderadar-test/build/generated-snippets/metrics/deltaTree/request-fields.adoc
asciidoctor: WARNING: restapi.adoc: line 641: include file not found: C:/Users/atanasov/workspace/CODE_RADAR/update-asciidoc/coderadar/coderadar-test/build/generated-snippets/metrics/deltaTree/response-fields.adoc
asciidoctor: WARNING: restapi.adoc: line 650: include file not found: C:/Users/atanasov/workspace/CODE_RADAR/update-asciidoc/coderadar/coderadar-test/build/generated-snippets/metrics/deltaTree/http-request.adoc
asciidoctor: WARNING: restapi.adoc: line 654: include file not found: C:/Users/atanasov/workspace/CODE_RADAR/update-asciidoc/coderadar/coderadar-test/build/generated-snippets/metrics/deltaTree/http-response.adoc
```

File: coderadar-test/src/main/asciidoc/restapi.adoc (hunk 2 above)
Comment (jo2) [reply coderadar-test/src/main/asciidoc/restapi.adoc]: sending encrypted password or the hash of the password both have advantages and disadvantages. sending it in clear publishes the password if encryption is cracked, sending it as a hash makes replay attacks possible if encryption is cracked..

**Comment by jo2** [comment]
Updated branch to current master. I don't get these warnings, could you please check again?

**Pull request merged**